i18n(es): Translated `from-nextjs.mdx` (#3124)
withastro/docs: src/content/docs/es/guides/migrate-to-astro/from-nextjs.mdx

            </>
        )
    }

    export default Component;
    ```
  </Fragment>
  <Fragment slot="astro">
    ```astro title="StarCount.astro"
    ---
    import Header from "./header";
    import Footer from "./footer";
    import "./layout.css";

    const res = await fetch('https://api.github.com/repos/withastro/astro')
    const json = await res.json()
    const message = json.message;
    const stars = json.stargazers_count || 0;
    ---
    <Header />
    <p class="banner">Astro has {stars} 🧑‍🚀</p>
    <Footer />

    <style>
      .banner {
        background-color: #f4f4f4; 
        padding: 1em 1.5em;
        text-align: center;
        margin-bottom: 1em;
      }
    <style>
    ```
  </Fragment>
</AstroJSXTabs>

### Migrando archivos de diseño

Puede resultarte útil comenzar convirtiendo tus diseños y plantillas de Next.js en [componentes de diseño de Astro](/es/core-concepts/layouts/).

Next tiene dos métodos diferentes para crear archivos de diseño, cada uno de los cuales maneja los diseños de manera diferente a Astro:

- El directorio `pages`

- [El directorio `/app` (en beta)](https://beta.nextjs.org/docs/routing/pages-and-layouts#layouts)

> Cada página de Astro requiere explícitamente que se incluyan las etiquetas `<html>`, `<head>`, y `<body>`, por lo que es comñun reutilizar un archivo de diseño en varias páginas. Astro utiliza [`<slot />`](/es/core-concepts/astro-components/#slots) para el contenido de la página, sin necesidad de importar ninguna declaración. Observa la estructura estándar de templating de HTML y el acceso directo al elemento `<head>`:  

Reviewer: ```suggestion
Cada página de Astro requiere explícitamente que se incluyan las etiquetas `<html>`, `<head>`, y `<body>`, por lo que es común reutilizar un archivo de diseño en varias páginas. Astro utiliza [`<slot />`](/es/core-concepts/astro-components/#slots) para el contenido de la página, sin necesidad de importar ninguna declaración. Observa la estructura estándar de templating de HTML y el acceso directo al elemento `<head>`:  
```

En Astro, **todo el contenido de tus páginas debe encontrarse dentro de `src/`**, ya sea en `src/pages` o en `src/content`.

#### Páginas de React

Tus páginas JSX existentes (`.js`) de Next deberán [convertirse de archivos JSX a páginas `.astro`](#consejos-convertir-archivos-jsx-a-archivos-astro). No puedes usar un archivo de página JSX existente en Astro

Estas [páginas `.astro`](/es/core-concepts/astro-pages/) deben estar ubicadas dentro de `src/pages/`  sus rutas de página se generarán automáticamente en función de su ruta de archivo.

#### Páginas de Markdown y MDX

Astro tiene soporte incorporado para Markdown y una integración opcional para archivos MDX. Puedes reutilizar cualquier archivo [Markdown y MDX](/es/guides/markdown-content/) existente, pero pueden requerir algunos ajustes en su frontmatter, como agregar [la propiedad especial de frontmatter `layout`de Astro](/es/core-concepts/layouts/#plantillas-de-markdownmdx). Ya no necesitarás crear manualmente páginas para cada ruta generada por Markdown. Estos archivos se pueden colocar dentro de`src/pages/` para aprovechar el enrutamiento automático basado en archivos.

> Alternativamente, puedes usar [colecciones de contenido](/es/guides/content-collections/) en Astro para almacenar y administrar tu contenido. Cuando forman parte de una colección, los archivos Markdown y MDX vivirán en carpetas dentro de `src/content/`.  Deberás recuperar el contenido tú mismo y [generar esas páginas dinámicamente](/es/guides/content-collections/#generating-routes-from-content).

Reviewer: ```suggestion
Alternativamente, puedes usar [colecciones de contenido](/es/guides/content-collections/) en Astro para almacenar y administrar tu contenido. Cuando forman parte de una colección, los archivos Markdown y MDX vivirán en carpetas dentro de `src/content/`. Deberás recuperar el contenido tú mismo y [generar esas páginas dinámicamente](/es/guides/content-collections/#generating-routes-from-content).
```

### Migrando Tests

> Como Astro produce HTML sin procesar, es posible escribir tests end-to-end utilizando la salida del paso de compilación. Cualquier test end-to-end escrito anteriormente  podría funcionar sin problemas si has podido coincidir con el marcado de tu sitio de Next. Bibliotecas de testing como Jest y React Testing Library pueden ser importadas y utilizadas en Astro para testear tus componentes de React.

Reviewer: ```suggestion
Como Astro produce HTML sin procesar, es posible escribir tests end-to-end utilizando la salida del paso de compilación. Cualquier test end-to-end escrito anteriormente podría funcionar sin problemas si has podido coincidir con el marcado de tu sitio de Next. Bibliotecas de testing como Jest y React Testing Library pueden ser importadas y utilizadas en Astro para poner a prueba tus componentes de React.
```
